src/lib/rules/http-compression/http-compression.ts

```diff
@@ -0,0 +1,673 @@
+
+                    // For anything else just flag it as disallowed.
+                    await context.report(resource, element, generateDisallowedCompressionMessage(encoding));
+                }
+            }
+
+            /*
+             * Special cases:
+             *
+             *  * SDCH (Shared Dictionary Compression over HTTP)
+             *    https://lists.w3.org/Archives/Public/ietf-http-wg/2008JulSep/att-0441/Shared_Dictionary_Compression_over_HTTP.pdf
+             *    Theoretically this should only happen if the user
+             *    agent advertises support for SDCH, but yet again,
+             *    server might be misconfigured.
+             *
+             *    For SDCH, the first response will not contain anything
+             *    special regarding the `content-encoding` header, however,
+             *    it will contain the `get-dictionary` header.
+             */
+
+            if (normalizeString(response.headers['get-dictionary'])) {
+                await context.report(resource, element, generateDisallowedCompressionMessage('sdch'));
+            }
+        };
+
+        const checkUncompressed = async (resource: string, element: IAsyncHTMLElement) => {
+
+            /*
+             * From: http://httpwg.org/specs/rfc7231.html#header.accept-encoding
+             *
+             *   " An "identity" token is used as a synonym for
+             *     "no encoding" in order to communicate when no
+             *     encoding is preferred.
+             *
+             *     ...
+             *
+             *     If no Accept-Encoding field is in the request,
+             *     any content-coding is considered acceptable by
+             *     the user agent. "
+             */
+
+            const { contentEncodingHeaderValue, rawResponse } = await getNetworkData(resource, { 'Accept-Encoding': 'identity' });
+
+            if (responseIsCompressed(rawResponse, contentEncodingHeaderValue)) {
+                await context.report(resource, element, generateCompressionMessage('', true, `for requests made with 'Accept-Encoding: identity'`));
+            }
+
+            if (contentEncodingHeaderValue) {
+                await context.report(resource, element, generateContentEncodingMessage('', true, `for requests made with 'Accept-Encoding: identity'`));
+            }
+        };
+
+        const isCompressibleAccordingToMediaType = (mediaType: string): boolean => {
+            if (!mediaType) {
+                return false;
+            }
+
+            /*
+             * The reason for doing the following is because
+             * `mime-db` is quite big, so querying it for
+             * eveything is expensive.
+             */
+
+            /*
+             * Check if the media type is one of the common
+             * ones for which it is known the response should
+             * be compressed.
+             */
+
+            if (isTextMediaType(mediaType) ||
+                ['image/x-icon',
```

Comment:
I think you should have these kind of arrays of the method as a constant.

Reply:
Fixed.